crates/consensus/src/validation.rs

```diff
@@ -656,6 +724,39 @@ mod tests {
             validate_block_standalone(&block, &chain_spec),
             Err(ConsensusError::WithdrawalIndexInvalid { .. })
         );
+
+        let (_, parent) = mock_block();
+        let mut provider = Provider::new(Some(parent.clone()));
+        // Withdrawal index should be 0 if there are no withdrawals in the chain
+        let block = create_block_with_withdrawals(&[1, 2, 3]);
+        provider.withdrawals_provider.expect_latest_withdrawal().return_const(Ok(None));
```

Comment:
pretty nice API :) I think we've done this manually in a lot of other places, maybe worth investigating as an alternative in our testing, cc @onbjerg @mattsse @rkrasiuk